tools/codegen/gen.py

```diff
@@ -327,7 +337,118 @@ def func(f: NativeFunction) -> Optional[str]:
         else:
             assert_never(target)
 
-    return func
+    def structured_func(g: NativeFunctionGroup) -> List[str]:
+        assert g.out is not None
+
+        # Illegal to not have dispatch for structured kernels
+        with native_function_manager(g.out):
+            assert g.out.dispatch is not None, "dispatch on structured kernels currently mandatory"
```

Comment:
Is there something preventing this currently? Looking at native_functions.yaml, I can see a few ops without dispatch keys that look like they could benefit from structured kernels (e.g. arccosh and arcsinh)

Reply:
This is a good question. I don't know the answer off the top of my head, I'll probably have to figure it out by trying to port arccosh.

Reply:
actually... maybe this will get fixed for free from your other PR? If we decide that any ops without dispatch entries will implicitly get a Math key.

Reply:
Well, we certainly don't have to assert dispatch is not None anymore :) But I'm still not sure what will happen with this logic when it is run on a "Math" kernel, since these generate slightly differently than normal kernels.